modules/findUnreachableRoutes.js

```diff
@@ -0,0 +1,42 @@
+import { createMemoryHistory } from 'history'
+import warning from './warning'
+
+const createLocation = createMemoryHistory().createLocation
+
+/**
+ * Checks a Route for the component that SHOULD be rendered against the
+ * component that WILL be rendered. If falsy, this indicates an
+ * unreachable route.
+ */
+function verifyRouteIsReachable(path, route, transitionManager) {
+  if (!path || !route) return
+
+  // strip vars so the path becomes routable
+  const cleanPath = path.replace(':', '').replace('*', 'splat')
+
+  transitionManager.match(createLocation(cleanPath), (_, _1, state) => {
```

Comment:
Update: Sat down and got some great feedback from @mjackson on this. He suggested putting the call to `checkForUnreachableRoutes` somewhere in the `createRoutes()` function: https://github.com/reactjs/react-router/blob/master/modules/RouteUtils.js#L30-L47

Will redo based on his feedback.